src/Isolinear.rei

```diff
@@ -1,47 +1,122 @@
-module Effect = Effect;
-module Store = Store;
-module Stream = Stream;
-module Updater = Updater;
+type dispatcher('msg) = 'msg => unit;
+type unsubscribe = unit => unit;
+
+module Stream: {
+  type t('msg);
+
+  let create: unit => (t('msg), dispatcher('msg));
+
+  let subscribe: (t('msg), dispatcher('msg)) => unsubscribe;
+  let connect: (dispatcher('msg), t('msg)) => unsubscribe;
+  let filterMap: (t('a), 'a => option('b)) => t('b);
+};
+
+module Effect: {
+  type t('msg);
+
+  let create: (~name: string, unit => unit) => t('a);
+  let createWithDispatch:
+    (~name: string, dispatcher('msg) => unit) => t('msg);
+
+  let none: t(_);
+  let batch: list(t('msg)) => t('msg);
+  let map: ('a => 'b, t('a)) => t('b);
+
+  let name: t(_) => string;
+  let run: (t('msg), dispatcher('msg)) => unit;
+};
+
+module Updater: {
+  type t('model, 'msg) = ('model, 'msg) => ('model, Effect.t('msg));
+
+  let ofReducer: (('model, 'msg) => 'model) => t('model, 'msg);
+  let combine: list(t('model, 'msg)) => t('model, 'msg);
+};
 
 module Sub: {
+  type t('msg);
+
+  let none: t('msg);
+  let batch: list(t('msg)) => t('msg);
+  let map: ('a => 'b, t('a)) => t('b);
+
+  module type S = {
+    type params;
+    type msg;
+
+    let create: params => t(msg);
+  };
+
   module type Config = {
     type params;
     type msg;
 
     // State that is carried by the subscription while it is active
     type state;
 
-    let subscriptionName: string;
-
-    let getUniqueId: params => string;
+    let name: string;
+    let id: params => string;
 
     let init: (~params: params, ~dispatch: msg => unit) => state;
     let update:
       (~params: params, ~state: state, ~dispatch: msg => unit) => state;
     let dispose: (~params: params, ~state: state) => unit;
   };
```

Comment:
I don't think `Config` is a great name for this. It's overly generic and you can see it causing ambiguity in `SubscriptionRunner` for example. Why was it renamed from `Provider`?

Reply:
I don't recall why I changed it - but `Provider` is fine with me.